server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/ObjectService.scala

```diff
@@ -213,7 +213,12 @@ class ObjectService(
           lockToPush <- previousMeta match {
             case Some(meta) =>
               checkLock(meta.lock, now, gsPath.bucketName)
-            case None => IO.raiseError(InvalidLock("Lock not found in local cache. You should acquireLock more often"))
+            case None =>
```

Comment:
I don't know best practices in Scala, but yielding `(lockMeta, generation)` from here would be clearer than having the logic in two places, then you just return `lockMeta, 0`

Reply:
it wouldn't make this cleaner since `checkLock(meta.lock, now, gsPath.bucketName)` is an `IO[Map[String, String]]` while generation is a pure value. So in order to yield `(lockMeta, generation)`, you need to lift `generation` into an `IO` first, which is unnecessary.